### PR TITLE
Add augmentations to RetinaNet overview guide

### PR DESCRIPTION
**Pull request by LukeWood**

### CODE REVIEW

**Comment by LukeWood** [comment]
Not quite ready for review yet, lets wait for Mosaic and perhaps RRC

**Comment by LukeWood** [comment]
@fchollet a PR to improve our OD scores!

**Comment by LukeWood** [comment]
@fchollet any blocker on this?

**Comment by LukeWood** [comment]
@fchollet let me know how this looks - it is significantly better results wise than the original (MaP 0.07-> 0.16)

File: guides/md/keras_cv/retina_net_overview.md [hunk 1]
```diff
@@ -301,7 +334,7 @@ model.save_weights(CHECKPOINT_PATH)
 
 <div class="k-default-codeblock">
 ```
-313/313 [==============================] - 145s 424ms/step - loss: 17.2746 - classification_loss: 10.0500 - box_loss: 6.8747 - val_Mean Average Precision: 0.0105 - val_Recall: 0.0744 - val_loss: 11.2469 - val_classification_loss: 5.2460 - val_box_loss: 5.6515 - val_regularization_loss: 0.3493 - lr: 0.0100
+313/313 [==============================] - 162s 481ms/step - loss: 12.5135 - classification_loss: 6.3680 - regularization_loss: 0.0000e+00 - box_loss: 6.1454 - val_Mean Average Precision: 0.0151 - val_Recall: 0.0432 - val_loss: 10.3468 - val_classification_loss: 5.1704 - val_box_loss: 5.1764 - val_regularization_loss: 0.0000e+00 - lr: 0.0100
```
Comment (fchollet) [inline guides/md/keras_cv/retina_net_overview.md:337]: Why is there an all-zero regularization loss?

File: guides/md/keras_cv/retina_net_overview.md [hunk 2]
```diff
@@ -90,6 +90,35 @@ def visualize_dataset(dataset, bounding_box_format):
 visualize_dataset(dataset, bounding_box_format="xywh")
 
 ```
+
+<div class="k-default-codeblock">
```
Comment (fchollet) [inline guides/md/keras_cv/retina_net_overview.md:94]: Please remove this block, it adds noise to the rendered page

**Review by fchollet — COMMENTED** [review]
Thanks!

File: guides/md/keras_cv/retina_net_overview.md (hunk 1 above)
Comment (LukeWood) [reply guides/md/keras_cv/retina_net_overview.md:337]: Hmm, I manually separated this out in the RetinaNet to ensure that it didn't leak into classification_loss or box_loss in the metrics.  Is this a bad UX?

Comment (fchollet) [reply guides/md/keras_cv/retina_net_overview.md:337]: But why is it 0? If there is no regularization loss, we should not display it

Comment (LukeWood) [reply guides/md/keras_cv/retina_net_overview.md:337]: Yeah, I'll fix this.  https://github.com/keras-team/keras-cv/issues/821

**Comment by LukeWood** [comment]
@fchollet the new guide looks WAY better!  You can actually see the importance of the prediction decoder, and everything else. 

PTAL

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
